pkg/v1/guestcluster/guestcluster.go

```diff
@@ -44,7 +44,9 @@ func (s *Service) NewG8sClient(ctx context.Context, clusterID, apiDomain string)
 	s.logger.LogCtx(ctx, "level", "debug", "message", "looking for certificate to connect to the guest cluster")
 
 	operatorCerts, err := s.certsSearcher.SearchClusterOperator(clusterID)
-	if err != nil {
+	if certs.IsTimeout(err) {
+		return nil, microerror.Maskf(notFoundError, "cluster-operator cert not found for cluster")
```

Comment:
We use a shorter 5 secs timeout rather than the default 90 secs and a not found error is returned.